pkg/apis/maistra/conversion/security.go

```diff
@@ -148,6 +148,86 @@ func populateSecurityValues(in *v2.ControlPlaneSpec, values map[string]interface
 			if err := setHelmStringValue(values, "global.caAddress", custom.Address); err != nil {
 				return err
 			}
+		case v2.CertificateAuthorityTypeCertManager:
+			if err := setHelmStringValue(values, "pilot.ca.implementation", string(security.CertificateAuthority.Type)); err != nil {
+				return fmt.Errorf("cert-manager ca config: failed converting CA implementation to helm: %s" ,err.Error())
+			}
+
+			CertManagerConf := security.CertificateAuthority.CertManager
+			if CertManagerConf == nil {
+				break
+			}
+
+			if err := setHelmStringValue(values, "global.caAddress", CertManagerConf.Address); err != nil {
+				return fmt.Errorf("cert-manager ca config: failed converting CA address to helm: %s" ,err.Error())
+			}
+
+			addEnvToComponent(in, "pilot", "ENABLE_CA_SERVER", "false")
+
+			extraArgs := []string{
+				"--tlsCertFile=/etc/cert-manager/tls/tls.crt",
+				"--tlsKeyFile=/etc/cert-manager/tls/tls.key",
+				"--caCertFile=/etc/cert-manager/tls/ca.crt",
+			}
+
+			suppliedRootConfigMap:= (CertManagerConf.RootCAConfigMapName != "")
+
+			if suppliedRootConfigMap {
+				extraArgs[2] = "--caCertFile=/etc/cert-manager/ca/root-cert.pem"
+			} else {
+				fmt.Printf(`Warning! using istiod-tls ca.crt key as root of trust for mesh,\
```

Comment:
this should probably use the logging facility. If we really want to warn the user though, then this should be moved into the ValidatingWebhook.

Reply:
yes totally :+1: